localads/campaign_serialization.cpp

```diff
@@ -0,0 +1,84 @@
+#include "localads/campaign_serialization.hpp"
+
+#include "coding/varint.hpp"
+#include "coding/byte_stream.hpp"
+
+#include <vector>
+#include <type_traits>
+#include <cstdint>
+
+
+namespace
+{
+enum class Version
+{
+  unknown = -1,
+  v1 = 0,  // March 2017 (Store featrues ids as varints, use two bytes for
+           // both days before expiration and icon id.)
+  latest = v1
+};
+
+template<typename ByteStream, typename T>
+void Write(ByteStream & s, T t)
+{
+  s.Write(&t, sizeof(T));
+}
+
+template<typename T, typename ByteStream>
+T Read(ByteStream & s)
+{
+  T t;
```

Comment:
Enable if or static assert that T is pod. Please, pass T by reference, not return by value.